instrumentation/kubernetes-client-7.0/src/test/groovy/KubernetesRequestUtilsTest.groovy

```diff
@@ -3,85 +3,89 @@
-    expect:
-    KubernetesResource.parseRegularResource(urlPath).apiGroup == apiGroup
-    KubernetesResource.parseRegularResource(urlPath).apiVersion == apiVersion
-    KubernetesResource.parseRegularResource(urlPath).resource == resource
-    KubernetesResource.parseRegularResource(urlPath).subResource == subResource
-    KubernetesResource.parseRegularResource(urlPath).namespace == namespace
-    KubernetesResource.parseRegularResource(urlPath).name == name
-
-    where:
-    urlPath                                                        | apiGroup     | apiVersion | resource      | subResource | namespace | name
-    "/apis/apps/v1/deployments"                                    | "apps"       | "v1"       | "deployments" | null        | null      | null
-    "/apis/apps/v1/namespaces/default/deployments"                 | "apps"       | "v1"       | "deployments" | null        | "default" | null
-    "/apis/apps/v1/namespaces/default/deployments/foo"             | "apps"       | "v1"       | "deployments" | null        | "default" | "foo"
-    "/apis/apps/v1/namespaces/default/deployments/foo/status"      | "apps"       | "v1"       | "deployments" | "status"    | "default" | "foo"
-    "/apis/example.io/v1alpha1/foos"                               | "example.io" | "v1alpha1" | "foos"        | null        | null      | null
-    "/apis/example.io/v1alpha1/namespaces/default/foos"            | "example.io" | "v1alpha1" | "foos"        | null        | "default" | null
-    "/apis/example.io/v1alpha1/namespaces/default/foos/foo"        | "example.io" | "v1alpha1" | "foos"        | null        | "default" | "foo"
-    "/apis/example.io/v1alpha1/namespaces/default/foos/foo/status" | "example.io" | "v1alpha1" | "foos"        | "status"    | "default" | "foo"
-  }
-
-  def "parsing kubernetes request verbs should work"(String httpVerb, boolean hasNamePathParam, boolean hasWatchParam, KubernetesVerb kubernetesVerb) {
-    expect:
-    KubernetesVerb.of(httpVerb, hasNamePathParam, hasWatchParam) == kubernetesVerb
-
-    where:
-    httpVerb | hasNamePathParam | hasWatchParam | kubernetesVerb
-    "GET"    | true             | false         | KubernetesVerb.GET
-    "GET"    | false            | true          | KubernetesVerb.WATCH
-    "GET"    | false            | false         | KubernetesVerb.LIST
-    "POST"   | false            | false         | KubernetesVerb.CREATE
-    "PUT"    | false            | false         | KubernetesVerb.UPDATE
-    "PATCH"  | false            | false         | KubernetesVerb.PATCH
-    "DELETE" | true             | false         | KubernetesVerb.DELETE
-    "DELETE" | false            | false         | KubernetesVerb.DELETE_COLLECTION
-  }
-}
+// WIP(anuraaga): The best way to support these sort of unit tests is to separate out library /
```

Comment:
Presumably the "best" way of handling these tests is to have separate library and agent instrumentation - library instrumentation tests handle any unit tests that require classes not available when running with agent.

I could temporarily add a unit tests test set for these, but want to be a bit lazy - there are very few projects this affected.